### DISCUSSION TITLE
Recursion limit exceeded in logging

### DISCUSSION THREAD

**Discussion by brianreinhold** [micropython/micropython#15106] [RP2040 / Pico]
Occasionally I get a recursion limit exceeded error in my code running on the PICO W. Unfortunately it is not systematically reproduceable. So far, it has always occurred in the logger from the logging module. I'm not sure why it seems to be limited to the logger. I have long since abandoned regular print statements but I do not recall that error ever happening then.

Has anyone else any experience with this error and is there an alternative logging approach that might not encounter this error so often?

**Comment by ctimmer** [comment]
Interesting module, may have to try it out.

I noticed the logging module can raise an exception. Could logging be raising an exception (e.g. file write) and your application is assuming it did something wrong and is attempting to log the error which causes logging to raise another exception and so on.

To debug I would put a "sys.exit()" in front of the "raise e" in the logging module to see where the application is at when and if the exception occurs. 

Curt
